src/content/blog/hybrid-rendering.mdx

```diff
@@ -0,0 +1,114 @@
+Astro's build process is able to check for the presence of an `export const prerender = true` statement to determine which pages should be prerendered.
+To do this efficiently, we rely on the wonderful [`es-module-lexer`](https://github.com/guybedford/es-module-lexer) library, which is also used internally in Node.
+
+```astro
+---
+// This route should be generated at build time!
+export const prerender = true
+
+const text = await fetch('https://example.com/').then((res) => res.text())
+---
+
+<article set:html={text} />
+```
+
+## Use cases for hybrid rendering
+
+Hybrid rendering unlocks a whole new set of possible use cases, but we wanted to highlight a few that we're particularly excited about.
+
+### Improve render performance of popular pages
+
+Consider a small/medium e-commerce site built with Astro, with thousands of different possible products and variations. If you were to use static site generation, you'd be stuck rebuilding your website every time one of these products changed or went out of stock.
+
+This kind of project calls for server-side rendering, which would allow your product pages to be generated fresh on every request. But now we have a new problem: your homepage is no longer static. Loading performance is still critical for driving sales, and your homepage doesn't have any dynamic content that actually requires it to be rebuilt fresh on every request.
+
+Hybrid rendering solves our performance problem by rendering the homepage to static HTML ahead-of-time, while keeping the rest of your website rendering on-demand.
+
+### Add an API to your existing static site
+
+As your static website grows, you may find the need to add a public API. Historically, adding an API to a static site has required host-specific configuration, but in Astro 2.0, we can add [Server Endpoints (API Routes)](https://docs.astro.build/en/core-concepts/endpoints/#server-endpoints-api-routes) without sacrificing the performance of our static pages.
+
+### Improve build performance of large sites
+
+Static generation can be difficult to scale to thousands of pages, since every page must be rendered during the build process. Server-side rendering defers page rendering until request-time, removing a potential bottleneck from your build process.
+
+When there are hundreds of routes generated by `getStaticPaths`, server-side rendering can lead to significant performance gains. In our internal testing, we saw build times improve by up to 30% by switching dynamic routes to server-side rendering.
+
+## Getting started with hybrid rendering
+
+Hybrid rendering is available today in Astro 2.0! You can head to our [server-side rendering guide](https://docs.astro.build/en/guides/server-side-rendering/) to learn more about Astro's SSR features. Astro's official adapters have also been updated to support hybrid rendering, so be sure to install the latest version of your [adapter](https://docs.astro.build/en/guides/server-side-rendering/#adding-an-adapter).
```

Comment:
can we add a reference to https://www.youtube.com/watch?v=AFP3ZHxO7Gg ? Happy to help wordsmith but would be great to help them promote it more